src/Compilers/Core/Portable/CodeGen/StateMachineStateDebugInfo.cs

```diff
@@ -0,0 +1,50 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Immutable;
+using System.Linq;
+
+namespace Microsoft.CodeAnalysis.CodeGen;
+
+internal readonly record struct StateMachineStateDebugInfo(int SyntaxOffset, int StateNumber);
```

Comment:
>record struct [](http://example.com/codeflow?start=18&length=13)

Are we using the generated `Equals()` method, etc. for this type or the type below? If not consider using `readonly struct` instead.

Reply:
We don't use equals, but record is shorter to write.

Reply:
It looks like switching to a `readonly struct` would only require adding fields and a constructor. I think we should switch these two types.

Reply:
Will change to readonly struct.